pkg/kubelet/status/status_manager.go

```diff
@@ -838,6 +838,12 @@ func (m *manager) canBeDeleted(pod *v1.Pod, status v1.PodStatus) bool {
 	if pod.DeletionTimestamp == nil || kubetypes.IsMirrorPod(pod) {
 		return false
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// Delay deletion of Pending pods with a finalizer until the phase is terminal.
```

Comment:
nit: Leave a doc comment here that it is expected that `generateAPIPodStatus` is expected to transition pending pods to Failed phase and let's link to a issue TODO to enable this more broadly for all Running phase pods as well. 

Let's also note that adding the `len(pod.Finalizers) > 0` is optimization to avoid additional pod status update if the pod will be deleted anyway.

Reply:
Done